Sources/TokamakStaticHTML/Views/Text/Text.swift

```diff
@@ -89,22 +89,44 @@ extension Font: StylesConvertible {
   }
 }
 
+private struct TextSpan: AnyHTML {
+  let content: String
+  let attributes: [String: String]
+
+  var innerHTML: String? { content }
+  var tag: String { "span" }
+}
+
 extension Text: AnyHTML {
   public var innerHTML: String? {
-    switch _TextProxy(self).storage {
+    let proxy = _TextProxy(self)
+    switch proxy.storage {
     case let .verbatim(text):
       return text
     case let .segmentedText(segments):
       return segments
-        .map(\.outerHTML)
+        .map {
+          TextSpan(content: $0.0.rawText,
+                   attributes: Self.attributes(from: $0.1,
+                                               environment: proxy.environment))
+            .outerHTML
+        }
         .reduce("", +)
     }
   }
 
   public var tag: String { "span" }
   public var attributes: [String: String] {
     let proxy = _TextProxy(self)
-    let isRedacted = proxy.redactionReasons.contains(.placeholder)
+    return Self.attributes(from: proxy.modifiers,
+                           environment: proxy.environment)
+  }
+}
+
+extension Text {
+  static func attributes(from modifiers: [_Modifier],
```

Comment:
<!--
  1 failure:  Function body sho...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: Sources/TokamakStaticHTML/Views/Text/Text.swift;
  Line: 127;
-->
- :no_entry_sign: Function body should span 50 lines or less excluding comments and whitespace: currently spans 54 lines (`function_body_length`)